### PR TITLE
PostgreSQL Flexible Server New API Swagger Changes

### PR DESCRIPTION
**Pull request by pradeepkintali**
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i>

### Contribution checklist:
- [ ] I commit to follow the [Breaking Change Policy](http://aka.ms/bcforapi) of “no breaking changes
- [ ] I have reviewed the [documentation](https://aka.ms/ameonboard) for the workflow.
- [ ] [Validation tools](https://aka.ms/swaggertools) were run on swagger spec(s) and errors have all been fixed in this PR. [How to fix?](https://aka.ms/ci-fix)

If any further question about AME onboarding or validation tools, please view the [FAQ](https://aka.ms/faqinprreview).

### ARM API Review Checklist
- [x] Ensure to check this box if one of the following scenarios meet updates in the PR, so that label “WaitForARMFeedback” will be added automatically to involve ARM API Review. Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs, all “removals” and “adding a new property” no more require ARM API review.
  - Adding new API(s)
  - Adding a new API version
  - Adding a new service

- [ ] If you are blocked on ARM review and want to get the PR merged with urgency, please get the ARM oncall for reviews (*RP Manifest Approvers* team under <ins>Azure Resource Manager service</ins>) from IcM and reach out to them. 

### Breaking Change Review Checklist 
If there are following updates in the PR, ensure to request an approval from API Review Board as defined in the [Breaking Change Policy](http://aka.ms/bcforapi). 

- [ ] Removing API(s) in stable version
- [ ] Removing properties in stable version
- [ ] Removing API version(s) in stable version
- [ ] Updating API in stable version with Breaking Change Validation errors
- [ ] Updating API(s) in preview over 1 year

Please follow the link to find more details on [PR review process](https://aka.ms/SwaggerPRReview).

### CODE REVIEW

File: specification/postgresql/resource-manager/Microsoft.DBforPostgreSQL/preview/2020-02-14-privatepreview/examples/ServerGet.json [hunk 1]
```diff
@@ -24,7 +24,15 @@
             "storageMB": 524288,
             "backupRetentionDays": 7
           },
-          "availabilityZone": "1"
+          "haEnabled": "Enabled",
+          "availabilityZone": "1",
+          "standByAvailabilityZone": "2",
+          "maintenanceWindow": {
+            "dayOfWeek": 0,
+            "startHour": 23,
```
Comment (akamath46297) [inline specification/postgresql/resource-manager/Microsoft.DBforPostgreSQL/preview/2020-02-14-privatepreview/examples/ServerGet.json:32]: 0?

Comment (pradeepkintali) [reply specification/postgresql/resource-manager/Microsoft.DBforPostgreSQL/preview/2020-02-14-privatepreview/examples/ServerGet.json]: Updated to 0

File: specification/postgresql/resource-manager/Microsoft.DBforPostgreSQL/preview/2020-02-14-privatepreview/postgresql.json [hunk 2]
```diff
@@ -1250,18 +1337,39 @@
       ],
       "description": "Represents a server."
     },
+    "MaintenanceWindow": {
```
Comment (Suna-MS) [inline specification/postgresql/resource-manager/Microsoft.DBforPostgreSQL/preview/2020-02-14-privatepreview/postgresql.json:1340]: How about the old definitions for CustomerMaintenanceWindow or CustomerMaintenanceWindowList? will you remove them?

Comment (pradeepkintali) [reply specification/postgresql/resource-manager/Microsoft.DBforPostgreSQL/preview/2020-02-14-privatepreview/postgresql.json:1340]: Removed the Unused entity definitions

File: specification/postgresql/resource-manager/Microsoft.DBforPostgreSQL/preview/2020-02-14-privatepreview/postgresql.json [hunk 3]
```diff
@@ -986,21 +877,24 @@
...
     },
-    "/subscriptions/{subscriptionId}/providers/Microsoft.DBForPostgreSql/checkNameAvailability": {
+    "/subscriptions/{subscriptionId}/providers/Microsoft.DBForPostgreSql/locations/{locationName}/checkVirtualNetworkSubnetUsage": {
       "post": {
         "tags": [
-          "CheckNameAvailability"
+          "VirtualNetworkSubnetUsage"
         ],
-        "operationId": "CheckNameAvailability_Execute",
+        "operationId": "VirtualNetworkSubnetUsage",
```
Comment (QingqingYuan2) [inline specification/postgresql/resource-manager/Microsoft.DBforPostgreSQL/preview/2020-02-14-privatepreview/postgresql.json:891]: Follow convention, this needs to be 'VirtualNetworkSubnetUsage_Execute'

File: specification/postgresql/resource-manager/Microsoft.DBforPostgreSQL/preview/2020-02-14-privatepreview/postgresql.json [hunk 4]
```diff
@@ -1164,10 +1250,13 @@
...
-        "standbyCount": {
-          "type": "integer",
-          "format": "int32",
-          "description": "stand by count value can be either 0 or 1"
+        "maintenanceWindow": {
+          "$ref": "#/definitions/MaintenanceWindow",
+          "description": "Maintenance window of a server."
+        },
+        "haEnabled": {
+          "type": "string",
```
Comment (QingqingYuan2) [inline specification/postgresql/resource-manager/Microsoft.DBforPostgreSQL/preview/2020-02-14-privatepreview/postgresql.json:1258]: Limit this to 'enabled' and 'disabled':
      "enum": [
        "Enabled",
        "Disabled"
      ],
      "x-ms-enum": {
        "name": "SslEnforcementEnum",
        "modelAsString": false
      }

**Review by QingqingYuan2 — APPROVED** [review]

File: specification/postgresql/resource-manager/Microsoft.DBforPostgreSQL/preview/2020-02-14-privatepreview/postgresql.json (hunk 3 above)
Comment (pradeepkintali) [reply specification/postgresql/resource-manager/Microsoft.DBforPostgreSQL/preview/2020-02-14-privatepreview/postgresql.json:891]: Done

File: specification/postgresql/resource-manager/Microsoft.DBforPostgreSQL/preview/2020-02-14-privatepreview/postgresql.json (hunk 4 above)
Comment (pradeepkintali) [reply specification/postgresql/resource-manager/Microsoft.DBforPostgreSQL/preview/2020-02-14-privatepreview/postgresql.json:1258]: Defined as enum to restrict the values.

**Review by majastrz — APPROVED** [review]
Signed off from ARM side.

**Comment by xccc-msft** [comment]
@pradeepkintali Can you please try to fix the CI errors? I cannot merge as the failures.

**Comment by pradeepkintali** [comment]
> @pradeepkintali Can you please try to fix the CI errors? I cannot merge as the failures.

Fixed the Lint errors.

**Comment by QingqingYuan2** [comment]
@xccc-msft, can we merge this PR? We're waiting for generating SDK.

**Comment by xccc-msft** [comment]
@QingqingYuan2  I don't have access to merge this as CI failure.

@akning-ms Can you please help? Thanks.

**Comment by akning-ms** [comment]
based on discussion in mail. it is private preview though it is public repo. so approve breaking change

**Pull request merged**